allennlp/common/from_params.py

```diff
@@ -128,23 +128,46 @@ def create_kwargs(
     signature = inspect.signature(constructor)
     kwargs: Dict[str, Any] = {}
 
+    parameters = dict(signature.parameters)
+
+    # First we check for the presence of a **kwargs parameter.  If we find one, we look in the
+    # superclass constructor, to see if there are arguments there that we should try to also
+    # constructor.
+    has_kwargs = False
+    for param in parameters.values():
+        if param.kind == param.VAR_KEYWORD:
+            has_kwargs = True
+    if has_kwargs:
+        # "mro" is "method resolution order".  The first one is the current class, the next is the
+        # first superclass, and so on.  Taking the first superclass should work in all cases that
+        # we're looking for here.
+        superclass = cls.mro()[1]
```

Comment:
Is this fragile if someone does:

```
class(OtherType, RegisterableType):
```

And `OtherType` also has a constructor? Equally I don't know a great way to improve it, maybe checking if the superclass inherits from Registrable or something?

Reply:
Yes, that would probably break this.  My vote is to wait to figure out how to handle that case until someone actually runs into it; seems pretty unlikely that that would happen.

Reply:
Maybe Multiple Inheritance is not what you want to handle here, but are you sure this inference shouldn't be recursive? What if the superclass constructor has `kwargs`?